ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto

```diff
@@ -24,12 +24,38 @@ service CommandService {
   // Errors:
   // - ``UNAUTHENTICATED``: if the request does not include a valid access token
   // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
-  // - ``NOT_FOUND``: if the request does not include a valid ledger id
-  // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
+  // - ``NOT_FOUND``: if the request does not include a valid ledger id or the submission rank or
+  //   deduplication period are too old.
+  // - ``INVALID_ARGUMENT`` if one of the following holds:
+  //   - The payload is malformed or is missing required fields.
+  //   - A party involved in the transaction is not known on the ledger.
+  //   - The interpreted command is inconsistent with the state of the ledger
+  //     If the submission led to a completion event, the error details' metadata key ``completion_offset``
+  //     contains the corresponding completion offset. This offset can be used as a submission rank in later calls.
+  //     If the completion event at the offset is a definite answer, the metadata key ''definite_answer``
+  //     is set to ``true``.
```

Comment:
I'd move this to a `FAILED_PRECONDITION` error, as it is the state of the system that makes this argument invalid.

Reply:
That's IMO part of cleaning up our error story. It has little to do with command deduplication as such. We currently do report such a case as `INVALID_ARGUMENT`.

Reply:
Add a `// TODO(error clean-up)` to error.proto